src/client/ua_client_connect.c

```diff
@@ -1050,11 +1069,14 @@ connectActivity(UA_Client *client) {
         /* Nothing to do if the connection has not opened fully */
     case UA_SECURECHANNELSTATE_CONNECTING:
     case UA_SECURECHANNELSTATE_CLOSING:
+    case UA_SECURECHANNELSTATE_HEL_SENT:
         return;
 
         /* Send HEL */
     case UA_SECURECHANNELSTATE_CONNECTED:
-        client->connectStatus = sendHELMessage(client);
+        if (!client->channel.isReverseConnect) {
```

Comment:
You don't need to test `isReverseConnect`.
That should only be set for the `UA_SECURECHANNELSTATE_REVERSE_CONNECTED` state.
The original code should work fine.

Reply:
Done